sycl/source/detail/device_impl.hpp

```diff
@@ -208,6 +230,30 @@ class device_impl : public std::enable_shared_from_this<device_impl> {
 
     // device_traits.def
 
+    CASE(info::device::device_type) {
+      using device_type = info::device_type;
+      // TODO: Are these enums the same?
+      switch (get_info_impl<ur_device_type_t, UR_DEVICE_INFO_TYPE>()) {
+      case UR_DEVICE_TYPE_DEFAULT:
+        return device_type::automatic;
+      case UR_DEVICE_TYPE_ALL:
+        return device_type::all;
+      case UR_DEVICE_TYPE_GPU:
+        return device_type::gpu;
+      case UR_DEVICE_TYPE_CPU:
+        return device_type::cpu;
+      case UR_DEVICE_TYPE_FPGA:
+        return device_type::accelerator;
+      case UR_DEVICE_TYPE_MCA:
+      case UR_DEVICE_TYPE_VPU:
+        return device_type::custom;
```

Comment:
I think there should have been a bug prior to this change in the imlementation of this info descriptor (we were essentially bit-casting between UR/SYCL RT enums).